src/software/sensor_fusion/sensor_fusion_test.cpp

```diff
@@ -1010,3 +1010,289 @@ TEST_F(SensorFusionTest, breakbeam_fail_test_ssl)
+{
+    std::vector<TbotsProto::RobotStatus> robot_status_msgs;
+    auto robot_msg = std::make_unique<TbotsProto::RobotStatus>();
+    robot_msg->set_robot_id(1);
+    robot_msg->add_error_code(TbotsProto::ErrorCode::NO_ERROR);
+    robot_msg->mutable_motor_status()->mutable_front_right()->add_motor_faults(
+        TbotsProto::MotorFault::DRIVER_OVERTEMPERATURE);
+    robot_status_msgs.push_back(*robot_msg);
+
+    auto robot_msg2 = std::make_unique<TbotsProto::RobotStatus>();
+    robot_msg2->add_error_code(TbotsProto::ErrorCode::NO_ERROR);
+    robot_msg2->set_robot_id(2);
+    robot_msg2->mutable_motor_status()->mutable_back_right()->add_motor_faults(
+        TbotsProto::MotorFault::DRIVER_OVERTEMPERATURE);
+    robot_status_msgs.push_back(*robot_msg2);
+
+    SensorProto sensor_msg;
+
+    auto ssl_wrapper_packet =
+        createSSLWrapperPacket(std::move(geom_data), initDetectionFrame());
+    // set vision msg so that world is valid
+    *(sensor_msg.mutable_ssl_vision_msg()) = *ssl_wrapper_packet;
+
+    sensor_fusion.processSensorProto(sensor_msg);
+
+    sensor_fusion.detectInjuredRobots(robot_status_msgs);
+
+    std::vector<Robot> injured_robots =
+        sensor_fusion.getWorld().value().friendlyTeam().getInjuredRobots();
+    EXPECT_EQ(injured_robots.size(), 2);
+
+    if (injured_robots.size() == 2)
+    {
+        EXPECT_EQ(injured_robots[0].id(), 1);
+        EXPECT_EQ(injured_robots[1].id(), 2);
+    }
+}
+
+TEST_F(SensorFusionTest, test_detect_back_left_motor_fault)
+{
+    std::vector<TbotsProto::RobotStatus> robot_status_msgs;
+    auto robot_msg = std::make_unique<TbotsProto::RobotStatus>();
+    robot_msg->set_robot_id(1);
+    robot_msg->add_error_code(TbotsProto::ErrorCode::NO_ERROR);
+    robot_msg->mutable_motor_status()->mutable_front_left()->add_motor_faults(
+        TbotsProto::MotorFault::DRIVER_OVERTEMPERATURE_PREWARNING);
+    robot_status_msgs.push_back(*robot_msg);
+
+    auto robot_msg2 = std::make_unique<TbotsProto::RobotStatus>();
+    robot_msg2->add_error_code(TbotsProto::ErrorCode::NO_ERROR);
+    robot_msg2->set_robot_id(2);
+    robot_msg2->mutable_motor_status()->mutable_back_left()->add_motor_faults(
+        TbotsProto::MotorFault::DRIVER_OVERTEMPERATURE_PREWARNING);
+    robot_status_msgs.push_back(*robot_msg2);
+
+    SensorProto sensor_msg;
+
+    auto ssl_wrapper_packet =
+        createSSLWrapperPacket(std::move(geom_data), initDetectionFrame());
+    // set vision msg so that world is valid
+    *(sensor_msg.mutable_ssl_vision_msg()) = *ssl_wrapper_packet;
+
+    sensor_fusion.processSensorProto(sensor_msg);
+
+    sensor_fusion.detectInjuredRobots(robot_status_msgs);
+
+    std::vector<Robot> injured_robots =
+        sensor_fusion.getWorld().value().friendlyTeam().getInjuredRobots();
+    EXPECT_EQ(injured_robots.size(), 2);
+
+    if (injured_robots.size() == 2)
+    {
+        EXPECT_EQ(injured_robots[0].id(), 1);
+        EXPECT_EQ(injured_robots[1].id(), 2);
+    }
+}
```

Comment:
these tests are all very similar. consider making this into a parametrized test (`TEST_P`)

example: https://github.com/UBC-Thunderbots/Software/blob/e0ca810ead609908d0344c3d74fa2561eba7fd74/src/software/logger/csv_sink_test.cpp#L36

Reply:
bump @Mr-Anyone